### PR TITLE
wallet-ext: export account private key

### PR DESCRIPTION
**Pull request by pchrysochoidis**
* button fix disabled
* Link component add color and weight variants
* PasswordInputDialog component 
  * verifies the password that was inserted by the user
  * when password is correct notifies the parent component with the password
  * if the parent returns a promise waits and shows a loader
* HideShowDisplayBox component 
  * initially the value is hidden and allows the user to reveal it
  * will be used to display the exported key
* add export private key to account in settings
* export and display the private key of an account
* export private key as hex and always 32 bytes 
  * makes it compatible with metamask

https://user-images.githubusercontent.com/10210143/219970956-7c930f87-14a0-44e9-b32c-e65af2ceef26.mov

Key now is formatted as hex starting with `0x` and always 32 bytes

https://user-images.githubusercontent.com/10210143/220103637-6d263325-f289-4c88-9945-1b7fed74c2ec.mov

closes APPS-289

### CODE REVIEW

**Review by kchalkias — COMMENTED** [review]
ooc, why export private key is useful for users in practice? I guess most (well, on the avg population only some) will probably understand their mnemonic, but private key can be confusing (for non-technical people). Is this common in other multi-account wallets too?

**Comment by pchrysochoidis** [comment]
> ooc, why export private key is useful for users in practice? I guess most (well, on the avg population only some) will probably understand their mnemonic, but private key can be confusing (for non-technical people). Is this common in other multi-account wallets too?

Metamask has a similar functionality, I see import/export as a way to move individual accounts between wallets, and not transfer all of them (mnemonic). But adds some extra complexity, cannot be recovered using mnemonic, another terms private key, differences on the encoding/format wallets use for the key etc

cc @bwann52

File: apps/wallet/src/ui/app/components/HideShowDisplayBox.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    value: string;
+    copiedMessage?: string;
+};
+
+export function HideShowDisplayBox({
+    value,
+    copiedMessage,
+}: HideShowDisplayBoxProps) {
+    const [valueHidden, setValueHidden] = useState(true);
+    const copyCallback = useCopyToClipboard(value, {
```
Comment (Jibz1) [inline apps/wallet/src/ui/app/components/HideShowDisplayBox.tsx:22]: I thought we did not allow `CopyToClipboard` because of security concerns.

**Comment by mystie711** [comment]
> ooc, why export private key is useful for users in practice? I guess most (well, on the avg population only some) will probably understand their mnemonic, but private key can be confusing (for non-technical people). Is this common in other multi-account wallets too?

This is a common use case from what i have seen empirically - delegating single/multiple active account management to someone else.

**Comment by pchrysochoidis** [comment]
@mystie711 issues/changes mentioned above are addressed now

Screenshot 2023-02-22 at 01 21 42

Screenshot 2023-02-22 at 01 10 37


https://user-images.githubusercontent.com/10210143/220496740-ad2291ba-fa91-4fd8-a77b-c413920ceb77.mov

File: apps/wallet/src/ui/app/components/HideShowDisplayBox.tsx (hunk 1 above)
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/components/HideShowDisplayBox.tsx:21]: That's a good point. Although the only way to practically use it is copy maybe we should hide the copy functionality. What do you think @mystie711

Comment (mystie711) [reply apps/wallet/src/ui/app/components/HideShowDisplayBox.tsx:21]: Slack thread conclusion for reference
https://mysten-labs.slack.com/archives/C042YF60NH2/p1677156381118669

**Review by mystie711 — APPROVED** [review]
looking good.

File: apps/wallet/src/ui/app/background-client/index.ts [hunk 2]
```diff
@@ -275,6 +275,43 @@ export class BackgroundClient {
         );
     }
 
+    public async verifyPassword(password: string) {
+        return await lastValueFrom(
```
Comment (Jibz1) [inline apps/wallet/src/ui/app/background-client/index.ts:279]: do we need await here

File: apps/wallet/src/ui/app/background-client/index.ts [hunk 3]
```diff
@@ -275,6 +275,43 @@ export class BackgroundClient {
...
+                    type: 'keyring',
+                    method: 'verifyPassword',
+                    args: { password },
+                })
+            ).pipe(take(1))
+        );
+    }
+
+    public async exportAccount(password: string, accountAddress: SuiAddress) {
+        return await lastValueFrom(
```
Comment (Jibz1) [inline apps/wallet/src/ui/app/background-client/index.ts:291]: same here.

**Comment by Jibz1** [comment]
except for nit with await LGTM

**Pull request merged**